### PR TITLE
T5661: Add show ssh dynamic-protection and show log ssh dynamic-protection

### PR DESCRIPTION
**Pull request by JeffWDH**
## Change Summary

Adds functionality to show IPs blocked by SSH dynamic-protection as well as the SSHGuard logs.
Manual backport of #2369

vyos-documentation PR 1127 has been submitted to update the documentation.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5661

## Proposed changes

Add:
show ssh dynamic-protection
show log ssh dynamic-protection

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: src/op_mode/ssh.py [hunk 1]
```diff
@@ -60,3 +61,40 @@ def show_fingerprints(raw: bool, ascii: bool):
...
+
+def show_dynamic_protection(raw: bool):
+    config = ConfigTreeQuery()
+    if not config.exists("service ssh dynamic-protection"):
+        raise vyos.opmode.UnconfiguredSubsystem("SSH server dynamic-protection is not enabled.")
+
+    attackers = []
+    try:
+        # IPv4
+        attackers = attackers + json.loads(cmd("sudo nft -j list set ip sshguard attackers"))["nftables"][1]["set"]["elem"]
```
Comment (c-po) [inline src/op_mode/ssh.py]: We try to not have `sudo` in the op mode scripts as long as this is not ultimately required. Please rather add the `sudo` call to the OP mode XML template.

File: op-mode-definitions/show-log.xml.in [hunk 2]
```diff
@@ -398,12 +398,20 @@
...
               <help>Show log for Secure Shell (SSH)</help>
             </properties>
             <command>journalctl --no-hostname --boot --unit ssh.service</command>
-          </leafNode>
+            <children>
+              <node name="dynamic-protection">
+                <properties>
+                  <help>Show dynamic-protection log</help>
+                </properties>
+                <command>journalctl --no-hostname -p info -t sshguard -o short</command>
```
Comment (c-po) [inline op-mode-definitions/show-log.xml.in]: Any particular reason to diverge from the usual path by specifying the systemd unit to read from?

Should be: `journalctl --no-hostname --boot --unit sshguard.service` instead

File: op-mode-definitions/show-ssh.xml.in [hunk 3]
```diff
@@ -7,6 +7,12 @@
           <help>Show SSH server information</help>
         </properties>
         <children>
+          <node name="dynamic-protection">
+            <properties>
+              <help>Show SSH server dynamic-protection blocked attackers</help>
+            </properties>
+            <command>${vyos_op_scripts_dir}/ssh.py show_dynamic_protection</command>
```
Comment (c-po) [inline op-mode-definitions/show-ssh.xml.in]: Add `sudo` here and see other comment about sudo below.

**Review by c-po — CHANGES_REQUESTED** [review]

File: src/op_mode/ssh.py (hunk 1 above)
Comment (JeffWDH) [reply src/op_mode/ssh.py]: I see calls to sudo in many other of the Python op_mode scripts... This script does more than just show dynamic-protection stuff, such as showing SSH public key fingerprints, which does not require root. I think limiting the use of root to where it is only needed (showing the tables) makes the most security sense. 

Also, this was accepted as is to 1.5, do we really need to change it? If so, it should be updated there as well.

File: op-mode-definitions/show-log.xml.in (hunk 2 above)
Comment (JeffWDH) [reply op-mode-definitions/show-log.xml.in]: This was accepted like this on the 1.5 commit, does it need to change? The output is functionally the same. If so, like the sudo change, 1.5 would need to be updated as well.

File: src/op_mode/ssh.py (hunk 1 above)
Comment (c-po) [reply src/op_mode/ssh.py]: We have a lot of code accepted in 1.4 and 1.5 that we later iterate on. No big deal.

I will append a commit with the changes to this PR here

File: op-mode-definitions/show-log.xml.in (hunk 2 above)
Comment (c-po) [reply op-mode-definitions/show-log.xml.in]: We have a lot of code accepted in 1.4 and 1.5 that we later iterate on. No big deal.

I will append a commit with the changes to this PR here

**Review by c-po — APPROVED** [review]

File: src/op_mode/ssh.py (hunk 1 above)
Comment (c-po) [reply src/op_mode/ssh.py]: The remaining calls to `sudo` are there b/c those scripts have yet most likely not been converted to the new style op-mode scripts.

**Review by sever-sever — APPROVED** [review]

**Pull request merged**